tools/harness-thci/OpenThread_BR.py

```diff
@@ -661,3 +664,54 @@ def powerUp(self):
     def forceSetSlaac(self, slaacAddress):
         print('forceSetSlaac %s' % slaacAddress)
         self.bash('sudo ip -6 addr add %s/64 dev wpan0' % slaacAddress)
+
+    # Override registerMulticast
+    @API
+    def registerMulticast(self, sAddr='ff04::1234:777a:1', timeout=300):
+        """subscribe to the given ipv6 address (sAddr) in interface and send MLR.req OTA
+
+        Args:
+            sAddr   : str : Multicast address to be subscribed and notified OTA.
+        """
+
+        if self.externalCommissioner is not None:
+            self.externalCommissioner.MLR([sAddr], timeout)
+            return True
+
+        # Register
+        cmd = 'sudo nohup ~/repo/openthread/tests/scripts/thread-cert/mcast6.py wpan0 %s' % sAddr
+        cmd = cmd + ' > /dev/null 2>&1 &'
+        self.bash(cmd)
+
+        # Add route
+        cmd = 'sudo ip -6 route add %s dev wpan0' % sAddr
```

Comment:
why do we need this route?

Reply:
Actually I misunderstood your comment here (R2) https://github.com/openthread/openthread/issues/7175#issuecomment-972613573

We don't need the route for the multicast destination, but I wonder if the default route needed for MATN-TC-02 would break other tests.

Reply:
Adding a default route for certification seems to be an over-kill. 

I think a simpler solution would be changing the echo mode from `OT_ICMP6_ECHO_HANDLER_DISABLED` to `OT_ICMP6_ECHO_HANDLER_ALL`, so that Thread stack will handle all `Ping Requests` without passing to Linux kernel, so that we don't rely on any kernel routes for responding to Ping Requests. 

Thoughts?
@edmont @bukepo @librasungirl

Reply:
@simonlingoogle I built the OTBR with you https://github.com/openthread/openthread/pull/7187 changes plus `OT_ICMP6_ECHO_HANDLER_ALL` enabled and the MATN-TC-02 test is passing now for FED and Router roles.

Reply:
Created https://github.com/openthread/openthread/pull/7233 for the echo handler setting.

Reply:
Okay, actually setting the `OT_ICMP6_ECHO_HANDLER_ALL` for the reference device only can't be the solution because the test should pass as DUT.

Reply:
For DUT, it can choose to use `OT_ICMP6_ECHO_HANDLER_ALL`, or it can choose to use OT_POSIX_INSTALL_EXTERNAL_ROUTES ?

So, it should still be simpler for a reference device to just use `OT_ICMP6_ECHO_HANDLER_ALL`?

Reply:
So do you think I should re-open #7233? In any case it's dependent on https://github.com/openthread/openthread/pull/7187

Reply:
I think it depends on #7187 no matter what, as long as `Border Routing` is enabled?

Reply:
I rebuild the reference device without `OT_ICMP6_ECHO_HANDLER_ALL` and with `-DOT_POSIX_INSTALL_EXTERNAL_ROUTES=ON` appended to the `OTBR_OPTIONS` in `otbr-setup.bash`. Replaced the `otbr-agent` binary in the Raspberry Pi and rerun the test, but it's failing with this configuration.

Reply:
Okay, I was actually testing without Border Routing enabled (`OTBR_BORDER_ROUTING`). When enabling it the device is replying properly over Ethernet to the multicast echo request received over Thread. So no manual route addition is required. The THCI works properly for the tests with the current changes.